lib/client/keyagent_test.go

```diff
@@ -125,104 +120,123 @@ func TestAddKey(t *testing.T) {
-	require.Equal(t, "teleport:"+s.username, teleportAgentKeys[1].Comment)
+	require.Equal(t, expectComment, teleportAgentKeys[1].Comment)
 
 	// check that we've loaded a cert as well as a private key into the system again
 	found := false
 	for _, sak := range systemAgentKeys {
-		if sak.Comment == "teleport:"+s.username && sak.Type() == "ssh-rsa" {
+		if sak.Comment == expectComment && sak.Type() == "ssh-rsa" {
 			found = true
 		}
 	}
 	require.True(t, found)
 	found = false
 	for _, sak := range systemAgentKeys {
-		if sak.Comment == "teleport:"+s.username && sak.Type() == ssh.CertAlgoRSAv01 {
+		if sak.Comment == expectComment && sak.Type() == ssh.CertAlgoRSAv01 {
 			found = true
 		}
 	}
 	require.True(t, found)
 
-	// unload all keys for this user from the teleport agent and system agent
-	err = lka.UnloadKey()
-	require.NoError(t, err)
 }
 
 // TestLoadKey ensures correct loading of a key into an agent. This test
 // checks the following:
 //   - Loading a key multiple times overwrites the same key.
+//   - Loading a key for separate teleport users/clusters does not overwrite existing keys.
 //   - The key is correctly loaded into the agent. This is tested by having
 //     the agent sign data that is then verified using the public key
 //     directly.
 func TestLoadKey(t *testing.T) {
 	s := makeSuite(t)
-
-	userdata := []byte("hello, world")
-
-	// make a new local agent
-	keystore, err := NewFSLocalKeyStore(s.keyDir)
-	require.NoError(t, err)
-	lka, err := NewLocalAgent(LocalAgentConfig{
-		Keystore:   keystore,
-		ProxyHost:  s.hostname,
-		Username:   s.username,
-		KeysOption: AddKeysToAgentAuto,
-	})
-	require.NoError(t, err)
-
-	// unload any keys that might be in the agent for this user
-	err = lka.UnloadKey()
-	require.NoError(t, err)
+	keyAgent := s.newKeyAgent(t)
 
 	// get all the keys in the teleport and system agent
-	teleportAgentKeys, err := lka.ExtendedAgent.List()
+	teleportAgentKeys, err := keyAgent.ExtendedAgent.List()
 	require.NoError(t, err)
-	teleportAgentInitialKeyCount := len(teleportAgentKeys)
-	systemAgentKeys, err := lka.sshAgent.List()
+	systemAgentKeys, err := keyAgent.sshAgent.List()
 	require.NoError(t, err)
-	systemAgentInitialKeyCount := len(systemAgentKeys)
 
-	// load the key to the twice, this should only
-	// result in one key for this user in the agent
-	err = lka.LoadKey(*s.key)
-	require.NoError(t, err)
-	err = lka.LoadKey(*s.key)
-	require.NoError(t, err)
+	// Create 3 separate keys, with overlapping user and cluster names
+	keys := []*Key{
+		s.key,
+		s.makeKey(t, s.key.Username, "other-proxy-host"),
+		s.makeKey(t, "other-user", s.key.ProxyHost),
+	}
 
-	// get all the keys in the teleport and system agent
-	teleportAgentKeys, err = lka.ExtendedAgent.List()
-	require.NoError(t, err)
-	systemAgentKeys, err = lka.sshAgent.List()
-	require.NoError(t, err)
+	// We should see two agent keys for each key added
+	agentsPerKey := 2
+	if runtime.GOOS == constants.WindowsOS {
```

Comment:
Can you even run our tests on Windows? 😅

Reply:
I'm not sure :grimacing: but I'll keep this here just in case / to clarify why we expect 2 agent keys per key.